scheduler/stack.go

```diff
@@ -17,8 +17,16 @@ const (
 	// serviceJobAntiAffinityPenalty but for batch type jobs.
 	batchJobAntiAffinityPenalty = 10.0
 
-	// previousFailedAllocNodePenalty is a scoring penalty for nodes that a failed allocation was previously run on
+	// previousFailedAllocNodePenalty is a scoring penalty for nodes
+	// that a failed allocation was previously run on
 	previousFailedAllocNodePenalty = 50.0
+
+	// skipScoreThreshold is a threshold used in the limit iterator to skip nodes
+	// that have a score lower than this.
+	skipScoreThreshold = -10.0
+
+	// maxSkip limits the number of nodes that can be skipped in the limit iterator
+	maxSkip = 3
```

Comment:
How were these consts chosen? Not sure if we need to document it or if the reasons are obvious if you have more experience in the scheduler than I do, but I'm curious. :)

Reply:
I chose the skip threshold based on the minimum possible penalty (from one job anti affinity collision). The maxSkip is somewhat arbitary, didn't want it to be too high, and wanted to limit the max number of nodes we examine. With these constants, it will examine upto a max of 5 nodes.